src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java

```diff
@@ -308,39 +313,51 @@ private void checkConditionalNotes(CompanyProfileApi companyProfileApi,
                                        BalanceSheet balanceSheet, SmallFullLinks smallFullLinks,
                                        String transactionId, String companyAccountsId) throws ServiceException {
 
-        if (isMultipleYearFiler(companyProfileApi)) {
-
-            if ((isDebtorsCurrentAmountNull(balanceSheet) || isDebtorsPreviousAmountNull(balanceSheet))
+        if ((isDebtorsCurrentAmountNull(balanceSheet) 
+                && isDebtorsPreviousAmountNull(balanceSheet))
                 && smallFullLinks.getDebtorsNote() != null) {
 
-                debtorsService.deleteDebtors(transactionId, companyAccountsId);
-            }
-        } else {
-            if ((isDebtorsCurrentAmountNull(balanceSheet) && smallFullLinks.getDebtorsNote() != null)) {
-                debtorsService.deleteDebtors(transactionId, companyAccountsId);
-            }
+            debtorsService.deleteDebtors(transactionId, companyAccountsId);
+        }
+        
+        if ((isStocksCurrentAmountNull(balanceSheet) 
+                && isStocksPreviousAmountNull(balanceSheet))
+                && smallFullLinks.getStocksNote() != null) {
+
+            stocksService.deleteStocks(transactionId, companyAccountsId);
         }
+        
     }
 
     private boolean isDebtorsCurrentAmountNull(BalanceSheet balanceSheet) {
-        Long currentDebtors =
-            Optional.of(balanceSheet)
+        return Optional.of(balanceSheet)
                 .map(BalanceSheet::getCurrentAssets)
                 .map(CurrentAssets::getDebtors)
                 .map(Debtors::getCurrentAmount)
-                .orElse(0L);
-
-        return currentDebtors.equals(0L);
+                .orElse(0L).equals(0L);
     }
 
     private boolean isDebtorsPreviousAmountNull(BalanceSheet balanceSheet) {
-        Long previousDebtors =
-            Optional.of(balanceSheet)
+        return Optional.of(balanceSheet)
                 .map(BalanceSheet::getCurrentAssets)
                 .map(CurrentAssets::getDebtors)
                 .map(Debtors::getPreviousAmount)
-                .orElse(0L);
+                .orElse(0L).equals(0L);
+    }
+    
+    private boolean isStocksCurrentAmountNull(BalanceSheet balanceSheet) {
```

Comment:
Are these method names a little misleading? A value of 0 would return true, though 0 wouldn't be considered the same as null. Not worth blocking the PR for, just a thought.